src/lib/network_pool/snark_pool_diff.ml

```diff
@@ -87,11 +88,19 @@ end)
           Ok data
     in
     match data with
-    | Stable.V1.Add_solved_work (work, p) ->
+    | Stable.V1.Add_solved_work (work, ({Priced_proof.proof; fee} as p)) ->
         let%bind () =
-          Pool.verify_and_act pool ~work:(work, p)
-            ~sender:(Envelope.Incoming.sender t)
+          let check () =
+            Pool.verify_and_act pool ~work:(work, p)
+              ~sender:(Envelope.Incoming.sender t)
+          in
+          match Pool.request_proof pool work with
+          | None ->
+              check ()
+          | Some {fee= {fee= prev; _}; _} ->
+              if Currency.Fee.( < ) prev fee.fee then
```

Comment:
you want `<=`, we should use the first one we get at a given fee.

Reply:
Oops. Thanks!